### ISSUE TITLE
[terra-form-select] MultiSelect and TagSelect onClick Option and Backspace Don't Handle AT Announcements Properly

### ISSUE THREAD

**Issue by dev-cprice** [cerner/terra-core#2590]
# Bug Report

## Description

Both multiselect variants of terra-form-select (Multi and Tag) do not announce on assistive technology that an option has been unselected using clicks, taps, or backspace, while unselecting an option with the `Enter` key does.

## Steps to Reproduce

For both [MultiSelect](https://engineering.cerner.com/terra-ui/components/terra-form-select/form-select/multi-select) and [TagSelect](https://engineering.cerner.com/terra-ui/components/terra-form-select/form-select/tag-select),

1. navigate to the component's page  and scroll to the examples (the first one is sufficient to demo)
2. turn on AT, e.g JAWS, mac VoiceOver, or ios VoiceOver (most notable on phones since there's no workflow for `ArrowUp`/`ArrowDown` and `Enter`)
3. Click (or tap) on an option to select, it'll say `{option.display} Selected`
4. Click or tap on on the same option, or press backspace, it will just say `{option.display}` or nothing

## Expected Behavior

When clicking on a selected option, it should AT announce `{option.display} Unselected` just like `ArrowUp`/`ArrowDown` and `Enter` workflow.

## Possible Solution

We just need to update the `visuallyHiddenComponent`'s `innerHTML`, identical to [here](https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/multiple/Menu.jsx#L373), but in the [MultiSelect's Menu.handleOptionClick](https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/multiple/Menu.jsx#L392) and [TagSelect's Menu.handleOptionClick](https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/tag/Menu.jsx#L337)

## Environment

* Component Name and Version: terra-form-select, MultiSelect and TagSelect
* Using Assistive Technology, eg. JAWS, mac VoiceOVer, or iOS VoiceOver

## @ Mentions

@nickpith @bjankord
